src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -442,53 +448,58 @@ main() {
     exit $ret
 }
 
-# On exit, for the Nextflow task sub-jobs
-nf_task_exit() {
-  ret=$?
-  if [ -f .command.log ]; then
-    dx upload .command.log --path "${cmd_log_file}" --brief --wait --no-progress || true
-  else
-    >&2 echo "Missing Nextflow .command.log file"
-  fi
-  # mark the job as successful in any case, real task
-  # error code is managed by nextflow via .exitcode file
-  if [ -z ${exit_code} ]; then export exit_code=0; fi
-
-  # Make sure that subjob with errorStrategy == terminate end in 'failed' state
-  wait_time=240
+wait_for_terminate_or_retry() {
   terminate_record=$(dx find data --name $DX_JOB_ID --path $DX_WORKSPACE_ID:/.TERMINATE --brief | head -n 1)
-  retry_record=$(dx find data --name $DX_JOB_ID --path $DX_WORKSPACE_ID:/.RETRY --brief | head -n 1)
-  if [ "$exit_code" -ne "0" ] && [ -n "${terminate_record}" ]; then
+  if [ -n "${terminate_record}" ]; then
     echo "Subjob exited with non-zero exit_code and the errorStrategy is terminate."
-    echo "Waiting for the headjob to kill the job tree..."
-    sleep $wait_time
+    echo "Waiting for the head job to kill the job tree..."
+    sleep $MAX_WAIT_AFTER_JOB_ERROR
     echo "This subjob was not killed in time, exiting to prevent excessive waiting."
-    exit
+    # Expected the job tree to be killed because of "terminate" strategy
+    # If not done already by Nextflow, will do it here
+    exit $exit_code
```

Comment:
This is my opinion about how to handle this situation; open to discussion. APPS-1821

Reply:
I added comment to the jira ticket, we can go through it during our daily if you want

Reply:
After discussion, we decided to remove any unrelated logical change from this PR and handle it as a separate Jira / PR. @mhrvol